server/tests/test_server.py

```diff
@@ -1,5 +1,64 @@
+from hmac import digest
+import pytest
+
 from app import __version__
+from app.genbank import digest_sequence
+from app.level import VectorLevel
 
 
 def test_version():
     assert __version__ == "0.1.0"
+
+
+def test_sequence_with_no_inserts():
+    with pytest.raises(ValueError) as e:
+        digest_sequence(VectorLevel.LEVEL0, "")
+
+
+def test_sequence_with_no_thing_in_the_middle():
+    with pytest.raises(ValueError) as e:
+        digest_sequence(
+            VectorLevel.LEVEL0, "GGTCTCGAGACC"
+        )  # No thing in the middle here
+
+
+def test_sequence_min_length():
+    assert digest_sequence(VectorLevel.LEVEL0, "GGTCTCXXXXXXGAGACC") == (
+        7,
+        7,
+        "",
+    )
+
+
+def test_single_character():
```

Comment:
Keep this test but name it better. Also re-create a test for a single nucleotide "gene" of interest and call it `test_single_character`.

Reply:
I've renamed it to test test_single_nucleotide_XXX.
XXX = level0 or level0_backbone or level1_backbone